### PR TITLE
Setting deployment as Paused: true initially and flipping it as part …

### PR DESCRIPTION
**Pull request by ewolinetz**
…of an update

Addresses https://jira.coreos.com/browse/LOG-234

### CODE REVIEW

File: pkg/k8shandler/desirednodestate.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ type desiredNodeState struct {
 	MasterNum          int32
 	DataNum            int32
 	EnvVars            []v1.EnvVar
+	Paused						 bool
```
Comment (richm) [inline pkg/k8shandler/desirednodestate.go]: formatting?

File: pkg/k8shandler/desirednodestate.go [hunk 2]
```diff
@@ -68,6 +69,7 @@ func constructNodeSpec(dpl *v1alpha1.Elasticsearch, esNode v1alpha1.Elasticsearc
 		Labels:             dpl.Labels,
 		MasterNum:          masterNum,
 		DataNum:            dataNum,
+		Paused:							true,
```
Comment (richm) [inline pkg/k8shandler/desirednodestate.go]: formatting?

Comment (ewolinetz) [reply pkg/k8shandler/desirednodestate.go]: yup, just ran `gofmt`

File: pkg/k8shandler/desirednodestate.go [hunk 3]
```diff
@@ -207,6 +210,35 @@ func (cfg *desiredNodeState) CreateOrUpdateNode(owner metav1.OwnerReference) err
 		if err != nil && !errors.IsAlreadyExists(err) {
 			return fmt.Errorf("Could not create node resource: %v", err)
 		}
+
+		// set it back to being paused
+		cfg.Paused = true
+		nretries := -1
+		if getErr := node.query(); getErr != nil {
```
Comment (richm) [inline pkg/k8shandler/desirednodestate.go]: I don't think you need to do the `node.query()` here if you are doing it in the retry loop

**Comment by ewolinetz** [comment]
Unfortunately, we need to set `Paused: false` when we create so that it'll do the initial roll out

File: pkg/k8shandler/desirednodestate.go [hunk 4]
```diff
@@ -207,6 +210,35 @@ func (cfg *desiredNodeState) CreateOrUpdateNode(owner metav1.OwnerReference) err
...
+			nretries++
+			if getErr := node.query(); getErr != nil {
+				logrus.Debugf("Could not get Elasticsearch node resource %v: %v", cfg.DeployName, getErr)
+				return getErr
+			}
+			dep, updateErr := node.constructNodeResource(cfg, metav1.OwnerReference{})
+			if updateErr != nil {
+				return fmt.Errorf("Could not construct node resource %v for update: %v", cfg.DeployName, updateErr)
+			}
+			logrus.Infof("Updating node resource %v", cfg.DeployName)
```
Comment (richm) [inline pkg/k8shandler/desirednodestate.go]: If this is a message that should be printed (as opposed to a debugging message) you should print it only once e.g.
```go
if nretries == 0 {
    logrus.Infof("Updating node resource %v", cfg.DeployName)
}
```
or something like that

File: pkg/k8shandler/desirednodestate.go [hunk 5]
```diff
@@ -239,6 +272,33 @@ func (cfg *desiredNodeState) CreateOrUpdateNode(owner metav1.OwnerReference) err
 			return fmt.Errorf("Error: could not update status for Elasticsearch %v after %v retries: %v", cfg.DeployName, nretries, retryErr)
 		}
 		logrus.Debugf("Updated Elasticsearch %v after %v retries", cfg.DeployName, nretries)
+
+		// set it back to being paused
+		cfg.Paused = true
+		if getErr := node.query(); getErr != nil {
```
Comment (richm) [inline pkg/k8shandler/desirednodestate.go]: same comments here as above about `node.query()` and the retry loop and the `Infof` message

File: pkg/k8shandler/desirednodestate.go (hunk 3 above)
Comment (ewolinetz) [reply pkg/k8shandler/desirednodestate.go]: good point, i'll remove it

**Pull request merged**